scripts/us_bjs/nps/preprocess_data.py

```diff
@@ -0,0 +1,222 @@
+    df["Count_Jurisdiction_PrivatelyOperatedOutOfStateFacility_Female"] = df["PVOTHF"]
+    df["Count_Jurisdiction_PrivatelyOperatedOutOfStateFacility_Male"] = df["PVOTHM"]
+    df["Count_Jurisdiction_PrivatelyOperatedOutOfStateFacility"] = df["PVOTHF"] + df["PVOTHM"]
+    df["Count_Jurisdiction_LocallyOperatedLocalFacility_Female"] = df["LFF"]
+    df["Count_Jurisdiction_LocallyOperatedLocalFacility_Male"] = df["LFM"]
+    df["Count_Jurisdiction_LocallyOperatedLocalFacility"] = df["LFF"] + df["LFM"]
+    df["Count_Jurisdiction_FederallyOperatedFacility_Female"] = df["FEDF"]
+    df["Count_Jurisdiction_FederallyOperatedFacility_Male"] = df["FEDM"]
+    df["Count_Jurisdiction_FederallyOperatedFacility"] = df["FEDF"] + df["FEDM"]
+    df["Count_Jurisdiction_StateOperatedOutOfStateFacility_Female"] = df["OTHSTF"]
+    df["Count_Jurisdiction_StateOperatedOutOfStateFacility_Male"] = df["OTHSTM"]
+    df["Count_Jurisdiction_StateOperatedOutOfStateFacility"] = df["OTHSTF"] + df["OTHSTM"]
+    df["Count_Custody_NotAUSCitizen_Female"] = df["NCITZTOTF"]
+    df["Count_Custody_NotAUSCitizen_Male"] = df["NCITZTOTM"]
+    df["Count_Custody_NotAUSCitizen"] = df["NCITZTOTF"] + df["NCITZTOTM"]
+    df["Count_Custody_NotAUSCitizen_MaxSentenceGreaterThan1Year_Female"] = df["NCITZGT1F"]
+    df["Count_Custody_NotAUSCitizen_MaxSentenceGreaterThan1Year_Male"] = df["NCITZGT1M"]
+    df["Count_Custody_NotAUSCitizen_MaxSentenceGreaterThan1Year"] = df["NCITZGT1F"] + df["NCITZGT1M"]
+    df["Count_Custody_NotAUSCitizen_MaxSentence1YearOrLess_Female"] = df["NCITZLE1F"]
+    df["Count_Custody_NotAUSCitizen_MaxSentence1YearOrLess_Male"] = df["NCITZLE1M"]
+    df["Count_Custody_NotAUSCitizen_MaxSentence1YearOrLess"] = df["NCITZLE1F"] + df["NCITZLE1M"]
+    df["Count_Custody_NotAUSCitizen_Unsentenced_Female"] = df["NCITZUNSF"]
+    df["Count_Custody_NotAUSCitizen_Unsentenced_Male"] = df["NCITZUNSM"]
+    df["Count_Custody_NotAUSCitizen_Unsentenced"] = df["NCITZUNSF"] + df["NCITZUNSM"]
+    df["Count_Custody_Under18_Female"] = df["CUSLT18F"]
+    df["Count_Custody_Under18_Male"] = df["CUSLT18M"]
+    df["Count_Custody_Under18"] = df["CUSLT18F"] + df["CUSLT18M"]
+    get_total_jurisdiction_columns(df)
+    
+def convert_geoId(fips_code):
+    """Creates geoId column"""
+    return 'geoId/' + str(fips_code).zfill(2)
+
+def convert_missing_value_to_nan(value):
+    """codes for missing values are always negative and actual data is always >= 0"""
+    if isinstance(value, int) and value < 0:
+        return float("nan")
+    else:
+        return value
+        
+def convert_nan_to_empty_cell(value):
+    if pd.isna(value):
+        return ''
+    else:
+        return value
+    
+def preprocess_df(raw_df):
+    """cleans raw_df
+
+    Args:
+        raw_data: raw data frame to be used as starting point for cleaning
+    """
+    df = raw_df.copy()
+    column_names = list(df.columns)
+    df['GeoId'] = df['STATEID'].apply(convert_geoId)
```

Comment:
Instead of modifying the same "df", should we add to a new "out_df" ? Then it seems we can do away with a separate STAT_VAR_NAMES list, and just use all "out_df" columns as the DESIRED_COLUMNS.